ntcore/src/main/java/edu/wpi/first/networktables/NetworkTableEntry.java

```diff
@@ -224,6 +224,26 @@ public String getString(String defaultValue) {
     return NetworkTablesJNI.getStringArray(m_handle, defaultValue);
   }
 
+  /**
+   * Checks if a data value is of a type that can be placed in a NetworkTable entry.
+   *
+   * @param data the data to check
+   * @return true if the data can be placed in an entry, false if it cannot
+   */
+  public static boolean isValidDataType(Object data) {
+    return data instanceof Number
+        || data instanceof Boolean
+        || data instanceof String
+        || data instanceof double[]
+        || data instanceof Double[]
+        || data instanceof Number[]
```

Comment:
Why do you need to check for double and then number?  Aren't all numbers doubles?

Reply:
No, although technically `Double[]` is a subclass of `Number[]`... but this is how NetworkTableEntry does the lookup for `setValue(Object)`, which I basically duplicated here